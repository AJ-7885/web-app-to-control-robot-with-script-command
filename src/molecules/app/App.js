import React from 'react';
import logo from '../../assets/logo.svg';

import { useRobotState } from './hooks';
import { Header, Footer, CommandInstruction, RobotLandSide } from './../../atoms';

import {
  Container,
  MainContainer,
  ActionContainer,
  CommandPanel,
  StartButton,
  CommandsInput
} from './App.style';

import {
  initialCommandsState,
  initialRobotPositionState,
  initialProcessQueue
} from './initialData';

function App() {
  const [
    commands,
    setCommands,
    robotPosition,
    processQueue,
    setProcessQueue
  ] = useRobotState(
    initialCommandsState,
    initialRobotPositionState,
    initialProcessQueue
  );

  function startHandler() {
    const commandListTemp = commands.map((command) => command.split(' '));
    setProcessQueue({
      ...processQueue,
      step: commands.length,
      commands: [...commandListTemp]
    });
  }

  const inputCommand = (newUserCommands) => {
    const newCommadlist = newUserCommands.split(/\r?\n/);
    setCommands(newCommadlist);
  };

  return (
    <Container>
      <Header logo={logo} />
      <MainContainer>
        <CommandPanel>
          <CommandsInput
            onChange={(e) => inputCommand(e.target.value)}
            value={commands.join('\r\n')}
            rows="10"
          />
          <StartButton onClick={startHandler}>Start</StartButton>
          <CommandInstruction />
        </CommandPanel>
        <RobotLandSide robotPosition={robotPosition} />
      </MainContainer>
      <ActionContainer />
      <Footer />
    </Container>
  );
}
export default App;
