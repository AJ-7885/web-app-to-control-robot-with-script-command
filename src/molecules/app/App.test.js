import React from 'react';
import renderer from 'react-test-renderer';
// import { mount } from 'enzyme/build';
import App from './App';

//TODO: Fix the issue of unit test for useCustomHook
// import { useRobotState } from './hooks';

// import {
//   initialCommandsState,
//   initialRobotPositionState,
//   initialProcessQueue
// } from './initialData';

// describe('useCustomHook', () => {
//   let results;
//   const renderHook = (hook) => {
//     function HookWrapper() {
//       results = hook();
//       return null;
//     }
//     mount(<HookWrapper />);
//     return results;
//   };

// it('works', () => {
//   renderHook(
//     useRobotState(
//       initialCommandsState,
//       initialRobotPositionState,
//       initialProcessQueue
//     )
//   )
// expect(results).toEqual(0);

// act(() => {
// });
// });
// });

test('renders correctly', () => {
  const component = renderer.create(<App />).toJSON();
  expect(component).toMatchSnapshot();
});
