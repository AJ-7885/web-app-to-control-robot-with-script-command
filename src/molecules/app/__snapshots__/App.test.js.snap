// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-Axmtr ipEZKn"
>
  <div
    className="sc-AxjAm gDOrEr"
  >
    <img
      alt="logo"
      className="sc-AxirZ ktnnZK"
      src="logo.svg"
    />
  </div>
  <div
    className="sc-AxmLO dXPOuP"
  >
    <div
      className="sc-fzozJi kkxoqs"
    >
      <textarea
        className="sc-fzoLsD eWPVGl"
        onChange={[Function]}
        rows="10"
        value="POSITION 1 1 EAST
FORWARD 3
WAIT
TURNAROUND
FORWARD 1
RIGHT
FORWARD 2"
      />
      <button
        className="sc-fzpans eqyVxq"
        onClick={[Function]}
      >
        Start
      </button>
      <div
        className="sc-AxiKw jjJxfu"
      >
        Commands: 
        <br />
        <br />
        1. FORWARD "NUMBER VALUE"
        <br />
        2. WAIT
        <br />
        3. POSITION "X value" "Y value" "direction (WEST,EAST,NORTH,SOUTH)"
        <br />
        4. LEFT , RIGHT , UP . DOWN
        <br />
        5. TURNAROUND
      </div>
    </div>
    <div
      className="sc-AxhCb CoMqJ"
    >
      <div
        className="sc-AxhUy iacnrW"
      >
        <img
          alt="robot"
          className="sc-AxgMl dTIslH"
          src="robot.png"
        />
      </div>
    </div>
  </div>
  <div
    className="sc-fzplWN ldxkaA"
  />
  <div
    className="sc-AxheI fUYOYB"
  />
</div>
`;
